### ISSUE TITLE
Identify and allow removing redundant braces

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#52565]
**Brief description:**

A feature I miss from ReSharper is the ability to quickly remove redundant braces from code.

For example, if an `if` statement is to be made unconditional, its inner block should be promoted. There's no quick gesture for this. I can quickly delete the `if (...)` line, then invoke a codefix on the remaining `{ ... }` (caret on the brace) to remove the brace and promote content.

**Languages applicable:**

C# only (I don't believe VB has an equivalent scoping construct).

**Code example that the analyzer should report:**

```c#
void M()
{
    { // caret here, ctrl+dot, "Remove redundant braces"
        Console.WriteLine("Hello");
    }
}
```

...produces...

```c#
void M()
{
    Console.WriteLine("Hello");
}
```

**Additional information:**

An analyzer here might be helpful, however I would likely demote it to suggestion as there are legitimate (uncommon) use cases for braces around a chunk of code, to create a lexical scope alone.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by Youssef1313** [comment]
Similar to #51668 (but not duplicate).
